Foundation/src/DateTimeFormat.cpp

```diff
@@ -150,11 +150,25 @@ bool DateTimeFormat::hasFormat(const std::string& fmt)
 
 bool DateTimeFormat::isValid(const std::string& dateTime)
 {
-	for (const auto& f : REGEX_LIST)
-	{
-		if (RegularExpression(*f).match(dateTime)) return true;
-	}
-	return false;
+
+        static const RegularExpression regs[] = {
+                RegularExpression(DateTimeFormat::ISO8601_REGEX),
+                RegularExpression(DateTimeFormat::RFC822_REGEX),
+                RegularExpression(DateTimeFormat::RFC1123_REGEX),
+                RegularExpression(DateTimeFormat::HTTP_REGEX),
+                RegularExpression(DateTimeFormat::RFC850_REGEX),
+                RegularExpression(DateTimeFormat::RFC1036_REGEX),
+                RegularExpression(DateTimeFormat::ASCTIME_REGEX),
+                RegularExpression(DateTimeFormat::SORTABLE_REGEX)
+        };
+		// make sure the regex list and the array of regexes are in sync
+		poco_assert((sizeof(regs) / sizeof(regs[0])) == REGEX_LIST.size());
```

Comment:
REGEX_LIST is used only in this place. I propose to remove it as a class member and have it as a static member in this function.

Reply:
I left it where it is because it's part of the public API of this class, unfortunately.

Wasn't sure if this would be 1.13.x or 1.14 and didn't want to make a breaking change just in case.

Reply:
@andrewauclair 1.14, you can break it